### PR TITLE
Add React and Mapbox boilerplate

### LINKED ISSUE #20: Add React + Mapbox boilerplate
As a frontend dev, I want to translate a component from Figma to React and manipulate it in the browser
- Use a React PWA [template](https://create-react-app.dev/docs/making-a-progressive-web-app)
- Use the Mapbox GL JS [SDK](https://docs.mapbox.com/help/tutorials/use-mapbox-gl-js-with-react/)

### PR DESCRIPTION
**Pull request by tedspare**
# Summary

- Added a React PWA template from `create-react-app`
- Chose to group TSX, styling, and testing files in folders by component but this could be discussed
- Added a sample Mapbox component
- Added a constants.ts for magic numbers
- Added a sample .env but the real deal will be shared on Slack

## Test Plan

- Ran using `yarn start`, opened Google Chrome to localhost:3000, and watched the console in Developer Tools

## Related Issues

Closes #20

### CODE REVIEW

File: frontend/src/serviceWorkerRegistration.ts [hunk 1]
```diff
@@ -0,0 +1,142 @@
+// This optional code is used to register a service worker.
```
Comment (JDziewonski98) [inline frontend/src/serviceWorkerRegistration.ts:1]: Is this an autogenerated file?

File: frontend/src/service-worker.ts [hunk 2]
```diff
@@ -0,0 +1,80 @@
+/// <reference lib="webworker" />
```
Comment (JDziewonski98) [inline frontend/src/service-worker.ts:1]: Is this autogenerated / should I understand this?

File: frontend/src/serviceWorkerRegistration.ts (hunk 1 above)
Comment (JDziewonski98) [inline frontend/src/serviceWorkerRegistration.ts:1]: Is this autogenerated / should I understand this?

**Comment by JDziewonski98** [comment]
I do not understand the service worker code tbh, but the rest looks good

**Comment by yi-fang-yuan** [comment]
I'm a bit confused by the serviceWorker.ts, is it for production only?

File: frontend/src/service-worker.ts (hunk 2 above)
Comment (tedspare) [reply frontend/src/service-worker.ts:1]: Yep this is autogenerated by `create-react-app` - not sure what it does specifically but service-worker is for the app to work offline

File: frontend/src/serviceWorkerRegistration.ts (hunk 1 above)
Comment (tedspare) [reply frontend/src/serviceWorkerRegistration.ts:1]: Same as above, autogenerated

**Comment by tedspare** [comment]
> I'm a bit confused by the serviceWorker.ts, is it for production only? @yi-fang-yuan 

For production! Service workers let us run the app offline (the defining factor of a PWA) and cache requests until connected. [Full explanation](https://stackoverflow.com/questions/49314386/what-is-service-worker-in-react-js).

**Comment by yi-fang-yuan** [comment]
then I think its fine! I will merge it.

**Pull request merged**